x-pack/plugins/infra/public/components/asset_details/tabs/overview/kpi_grid.tsx

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React from 'react';
+
+import { EuiFlexGroup, EuiFlexItem, EuiSpacer } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { type KPIChartProps, Tile } from './tile';
+import { TOOLTIP } from './translations';
+import type { KPIProps } from './overview';
+
+const KPI_CHARTS: Array<Omit<KPIChartProps, 'loading' | 'subtitle'>> = [
+  {
+    type: 'cpuUsage',
+    trendLine: true,
+    backgroundColor: '#F1D86F',
+    title: i18n.translate('xpack.infra.hostsViewPage.metricTrend.cpuUsage.title', {
+      defaultMessage: 'CPU Usage',
+    }),
+    toolTip: TOOLTIP.cpuUsage,
+  },
+  {
+    type: 'normalizedLoad1m',
+    trendLine: true,
+    backgroundColor: '#79AAD9',
+    title: i18n.translate('xpack.infra.hostsViewPage.metricTrend.normalizedLoad1m.title', {
+      defaultMessage: 'Normalized Load',
+    }),
+    toolTip: TOOLTIP.rx,
+  },
+  {
+    type: 'memoryUsage',
+    trendLine: true,
+    backgroundColor: '#A987D1',
+    title: i18n.translate('xpack.infra.hostsViewPage.metricTrend.memoryUsage.title', {
+      defaultMessage: 'Memory Usage',
+    }),
+    toolTip: i18n.translate('xpack.infra.hostsViewPage.metricTrend.memoryUsage.tooltip', {
+      defaultMessage: 'Main memory usage excluding page cache.',
+    }),
+  },
+  {
+    type: 'diskSpaceUsage',
+    trendLine: true,
+    backgroundColor: '#F5A35C',
+    title: i18n.translate('xpack.infra.hostsViewPage.metricTrend.diskSpaceUsage.title', {
+      defaultMessage: 'Disk Space Usage',
+    }),
+    toolTip: TOOLTIP.tx,
+  },
+];
+
+export interface KPIGridProps extends KPIProps {
+  nodeName: string;
+}
+
+export const KPIGrid = ({ nodeName, dataView, dateRange }: KPIGridProps) => {
```

Comment:
I've noticed that the KPIGrid re-renders when switching tabs. Charts are heavy and can degrade performance. Let's wrap this in a `React.memo`

<img width="429" alt="image" src="https://github.com/elastic/kibana/assets/2767137/b72fefb2-0d8e-40e7-bce6-c7e69126fcc1">

Reply:
Good catch, thanks. Added 🙂